### PR TITLE
Work history updates

### PR DESCRIPTION
**Pull request by adamsilver**
- move interviews lower down
- move work history above degrees
- mark roles relevant to teaching (instead of worked with children)
- mark approximate dates
- mark gaps
- handle when no unpaid experience added
- improve generators
- move safeguarding and disability to bottom

### CODE REVIEW

File: app/data/generators/school-experience.js [hunk 1]
```diff
@@ -1,18 +1,22 @@
...
-  })
+const faker = require('faker')
+faker.locale = 'en_GB'
 
-  const count = faker.random.number({ min: 1, max: 4 })
-  const items = []
-  for (var i = 0; i < count; i++) {
-    items.push(item(faker))
+module.exports = () => {
+  var hasExperience = faker.helpers.randomize([true, false])
```
Comment (simonwhatley) [inline app/data/generators/school-experience.js:5]: This should be a `const`.

File: app/views/_includes/applications/work-history.njk [hunk 2]
```diff
@@ -1,25 +1,81 @@
-{% set workHistory = application.workHistory
-  | toArray
-  | sort(attribute="startDate")
+{% set workHistoryItems = application.workHistory.items
+  | sort(attribute="startDate") | reverse
 %}
-{% if workHistory.length >= 1 %}
+
+{% if workHistoryItems.length >= 1 %}
```
Comment (simonwhatley) [inline app/views/_includes/applications/work-history.njk:5]: You don't need the operator here, so can remove`>= 1`

File: app/views/applications/show.njk [hunk 3]
```diff
@@ -159,49 +142,28 @@
...
+      {{ govukSummaryList({
+        rows: [
+          {
             key: {
-              text: "Do you have any work history?"
+              text: "Do you have any interview needs?"
             },
             value: {
-              html: "No, I do not have any work history"
+              text: "Yes" if application.interviewNeeds.response else "No"
```
Comment (simonwhatley) [inline app/views/applications/show.njk]: If `application.interviewNeeds.response` is either "Yes" or "No", do you need:

```
"Yes" if application.interviewNeeds.response else "No"
```

or just

```
application.interviewNeeds.response
```

**Review by simonwhatley — CHANGES_REQUESTED** [review]

**Review by simonwhatley — APPROVED** [review]

File: app/views/_includes/applications/work-history.njk (hunk 2 above)
Comment (adamsilver) [reply app/views/_includes/applications/work-history.njk:5]: I know but I think it's clearer to read for people who don't know that 0 is falsy

**Pull request merged**